TezosKit/Client/TezosNodeClient.swift

```diff
@@ -427,44 +442,50 @@ public class TezosNodeClient: AbstractClient {
     _ operations: [Operation],
     source: String,
     keys: Keys,
-    completion: @escaping (String?, Error?) -> Void
+    completion: @escaping (Result<String, TezosKitError>) -> Void
   ) {
-    getMetadataForOperation(address: source) { [weak self] operationMetadata in
-      guard let self = self,
-            let operationMetadata = operationMetadata else {
-        let error = TezosKitError(kind: .unknown, underlyingError: nil)
-        completion(nil, error)
+    getMetadataForOperation(address: source) { [weak self] result in
+      guard let self = self else {
         return
       }
 
-      // Determine if the address performing the operations has been revealed. If it has not been,
-      // check if any of the operations to perform requires the address to be revealed. If so,
-      // prepend a reveal operation to the operations to perform.
-      var mutableOperations = operations
-      if operationMetadata.key == nil && operations.first(where: { $0.requiresReveal }) != nil {
-        let revealOperation = RevealOperation(from: source, publicKey: keys.publicKey)
-        mutableOperations.insert(revealOperation, at: 0)
-      }
-      let operationPayload =
-        self.createOperationPayload(operations: mutableOperations, operationMetadata: operationMetadata)
-
-      self.forgeOperation(
-        operationPayload: operationPayload,
-        operationMetadata: operationMetadata
-      ) { [weak self] result, error in
-        guard let self = self,
-              let result = result else {
-          completion(nil, error)
-          return
+      switch result {
+      case .failure(let error):
+        completion(.failure(error))
+      case .success(let operationMetadata):
+        // Determine if the address performing the operations has been revealed. If it has not been,
+        // check if any of the operations to perform requires the address to be revealed. If so,
+        // prepend a reveal operation to the operations to perform.
+        var mutableOperations = operations
```

Comment:
Too much nesting, move to a helper method

Reply:
Done